### PR TITLE
Fold stellar/horizon into stellar/go/services/horizon

### PR DESCRIPTION
**Pull request by nikhilsaraf**
**Notes for the current commit:**
1. `stellar/go/services/horizon/scripts/run_tests.bash` passes, and the horizon server builds and serves on `localhost:8000`

2. I've updated `stellar/go/services/horizon/README.md` to link to the `stellar/go/releases` page so these download links are currently broken. We should backfill these links by creating new releases, or update these links to point to the old files in `stellar/horizon/releases` until we decide to push out a new horizon release.

3. The horizon docs now point to filing all issues to the “go” monorepo (`stellar/go/issues`) so we should expect that change in behavior from the community down the road.

4. The only bash script I've updated is `stellar/go/services/horizon/scripts/run_tests.bash`; all other scripts in `stellar/go/services/horizon/scripts` are currently broken because they depend on gb. We should work on fixing these scripts as a separate commit if we can tolerate having them broken for a short window — _I can work on fixing them after this commit_.

5. I will also need to port any changes made to horizon that were made after I started my changes, there's 2 so far I think  — _I can include these changes after this commit is approved_.

**Changes needed in other repos to support this move:**
1. Need to update [https://github.com/stellar/developers/blob/master/repos.json](https://github.com/stellar/developers/blob/master/repos.json) to point to the new url — _I will work on this after this commit_. (UPDATE: see [https://github.com/stellar/developers/pull/88](https://github.com/stellar/developers/pull/88))

2. I need to add a deprecated message to the old HORIZON repo’s README.md with a pointer to the new location — _I can work on this after this commit_. (UPDATE: see [https://github.com/stellar/horizon/pull/407](https://github.com/stellar/horizon/pull/407))

3. UPDATE: stellar/docs needs to be updated with the new url (see [https://github.com/stellar/docs/pull/233](https://github.com/stellar/docs/pull/233))

### CODE REVIEW

**Comment by nikhilsaraf** [comment]
had not included the `/tree/master` portion for web-based github urls so fixed that in commit no. 4

**Comment by nullstyle** [comment]
Looks like we're getting failures in Travis when running the test suite.

**Comment by nikhilsaraf** [comment]
fixed a dependency issue (in commit 5) by picking the older option between stellar/go and stellar/horizon for goji.io, will address the remaining test failures to resolve the build.

File: .travis.yml [hunk 1]
```diff
@@ -2,21 +2,30 @@ language: go
...
+  postgresql: '9.4'
 go:
 - '1.6'
 - '1.7'
 - '1.8'
 - tip
 env:
-- GO15VENDOREXPERIMENT="1"
+  global:
+  - GO15VENDOREXPERIMENT="1"
```
Comment (nullstyle) [inline .travis.yml]: We don't need this anymore...  go's vendor functionality is standard now on all of our supported versions (1.7, 1.8, 1.9)

File: .travis.yml [hunk 2]
```diff
@@ -2,21 +2,30 @@ language: go
...
+  - DATABASE_URL=postgres://localhost:5432/horizon_test?sslmode=disable&user=postgres
```
Comment (nullstyle) [inline .travis.yml]: These environment variables (`DATABASE_URL` and `STELLAR_CORE_DATABASE_URL`)  should not be needed to run our tests in the monorepo.  Before we merge I think we should make the necessary changes to excise the need for their existence from the test suite.

File: .travis.yml [hunk 3]
```diff
@@ -2,21 +2,30 @@ language: go
...
+  - STELLAR_CORE_DATABASE_URL=postgres://localhost:5432/stellar-core_test?sslmode=disable&user=postgres
 install:
 - wget https://github.com/Masterminds/glide/releases/download/v0.12.3/glide-v0.12.3-linux-amd64.tar.gz
 - tar -xzvf glide-v0.12.3-linux-amd64.tar.gz
 - cp linux-amd64/glide $GOPATH/bin
 before_script: 
+- psql -c 'create database "horizon_test";' -U postgres
```
Comment (nullstyle) [inline .travis.yml]: likewise to the environment variables I think we should remove the static test databases from horizon's test suite before we merge. If possible, switch to the `dbtest.Postgres()` system we talked about.

File: support/scripts/run_tests [hunk 4]
```diff
@@ -1,7 +1,7 @@
 #! /bin/bash
 set -e
 
-TESTS=$(glide novendor | sed 's/^\./github.com\/stellar\/go/g')
+TESTS=$(glide novendor | grep "./services/" | sed 's/^\./github.com\/stellar\/go/g')
```
Comment (nullstyle) [inline support/scripts/run_tests]: I don't believe this is correct:  This script should be testing everything in the monorepo but horizon.  The command you've added here encodes "only run the services tests", right?

**Comment by nikhilsaraf** [comment]
Looks like the build is passing. This leaves the following changes which I plan to make once this PR is merged to fully complete the changes needed in this repo:
1. clean up the bash scripts in services/horizon/scripts -- this will unblock deployments for horizon since build_artifacts.bash is one of those scripts
2. port any changes that were made since I started my port (1 change)
3. convert tests to use the Postgres() call instead of the Start() call to parallelize the horizon tests and use the existing test script in .travis.yml

These are a lot of commits. Do you want me to clean them up into a fewer number of commits (or one commit) and re-submit with this finalized version?
@nullstyle

File: services/horizon/scripts/build_artifacts.bash [hunk 5]
```diff
@@ -1,7 +1,7 @@
 #! /usr/bin/env bash
 set -e
 
-DIST="dist"
+DIST="$GOPATH/dist"
```
Comment (nikhilsaraf) [inline services/horizon/scripts/build_artifacts.bash]: tested these locally and they work (except dev.bash)

all these follow the same pattern:
- add the `$GOPATH`
- convert `gb build [-flags]` to `go install [-flags] github.com/stellar/go/services/horizon/cmd/horizon`
- convert `gb test X` to `go test X`
- convert `gb generate X` to `go generate X`

File: services/horizon/scripts/load_scenario.bash [hunk 6]
```diff
@@ -3,8 +3,8 @@
 set -e
 
 SCENARIO=$1
-CORE_SQL=./src/github.com/stellar/go/services/horizon/test/scenarios/$SCENARIO-core.sql
-HORIZON_SQL=./src/github.com/stellar/go/services/horizon/test/scenarios/$SCENARIO-horizon.sql
+CORE_SQL=$GOPATH/src/github.com/stellar/go/services/horizon/test/scenarios/$SCENARIO-core.sql
+HORIZON_SQL=$GOPATH/src/github.com/stellar/go/services/horizon/test/scenarios/$SCENARIO-horizon.sql
```
Comment (nullstyle) [inline services/horizon/scripts/load_scenario.bash]: Unfortunately, GOPATH isn't guaranteed to be single-valued, so we can't use it directly.  You might look into using a script-relative path, the equivalent of `../../test/scenarios/$SCENARIO-core.sql`.

File: services/horizon/test/db/main.go [hunk 7]
```diff
@@ -5,8 +5,6 @@ package db
 
 import (
 	"log"
-	"os"
```
Comment (nullstyle) [inline services/horizon/test/db/main.go]: You might be interested in leveraging https://godoc.org/golang.org/x/tools/cmd/goimports  to help avoid this in the future

File: services/horizon/scripts/load_scenario.bash (hunk 6 above)
Comment (nikhilsaraf) [reply services/horizon/scripts/load_scenario.bash]: ok, converted all scripts to use a script-relative path, and created a new variable $GOTOP in the scripts where it was needed

File: services/horizon/test/db/main.go (hunk 7 above)
Comment (nikhilsaraf) [reply services/horizon/test/db/main.go]: thanks, will look into integrating this into my IDE.

**Comment by nikhilsaraf** [comment]
updated the scripts. let me know if there's anything else. I'm looking into any changes needed in the stellar/docs repo.

**Pull request merged**